### PR TITLE
server: refresh push context after cache sync

### PR DESCRIPTION
**Pull request by ramaraochavali**
As per discussion [here](https://github.com/istio/istio/pull/20296#discussion_r368355918)
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by ramaraochavali** [comment]
@hzxuzhonghu @howardjohn PTAL

**Comment by hzxuzhonghu** [comment]
@ramaraochavali This will not trigger any push when refreshing push context, right? Almost equals without any modification(without #20296).

**Comment by ramaraochavali** [comment]
@hzxuzhonghu yes, this will not trigger any push to the local proxy that is connected. I am assuming there will be configupdates that might trigger a push? Or should I trigger a `configupdate` to be on safe side?

**Comment by ramaraochavali** [comment]
@hzxuzhonghu Actually I think if we call push here, it does both - it updates the global push context and also pushes the latest config to the local proxy. I think it is safe - WDYT?

File: pilot/pkg/bootstrap/server.go [hunk 1]
```diff
@@ -292,9 +292,10 @@ func (s *Server) Start(stop <-chan struct{}) error {
 		return fmt.Errorf("failed to sync cache")
 	}
 
-	// Notify Server readiness to Discovery Server, so that it can start accepting connections.
-	log.Infof("All caches have been synced up, notifying discovery server")
-	s.EnvoyXdsServer.OnServerReady()
+	// Trigger a push, so that the global push context is updated with the new config and Pilot's local Envoy
+	// also is updated with new config.
+	log.Infof("All caches have been synced up, triggering a push")
+	s.EnvoyXdsServer.Push(&model.PushRequest{Full: true})
```
Comment (hzxuzhonghu) [inline pilot/pkg/bootstrap/server.go]: Correct

**Pull request merged**